lnd.go

```diff
@@ -618,6 +623,12 @@ func Main(cfg *Config, lisCfg ListenerCfg, interceptor signal.Interceptor) error
 			// The channel is buffered by one element so writing
 			// should not block here.
 			walletInitParams.MacResponseChan <- adminMacBytes
+
+			for _, lis := range grpcListeners {
```

Comment:
I think we should also close the mac channels after the `if !cfg.NoMacaroons {` block is over (where we also do `close(walletInitParams.MacResponseChan)`) to signal any listener that nothing is coming over that channel anymore. Especially for the case where no macaroons are used.